### ISSUE TITLE
Setting 'nofixendofline' and 'endoffile' and writing DOS format file add CTRL-Z characters to the beginning of the file

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#11469]
### Steps to reproduce

1. Run `vim --clean`
2. Source the following file:
```vim
call writefile(['aaa', 'bbb'], 'Xendoffile')
edit Xendoffile
set fileformat=dos nofixendofline endoffile
write
edit
```
3. Two CTRL-Z characters are added to the beginning of the file

### Expected behaviour

What 'endoffile' is documented to do

### Version of Vim

9.0.820

### Environment

Operating system: Arch Linux
Terminal: kitty
Value of $TERM: xterm-kitty
Shell: fish

**Comment by k-takata** [comment]
Should be fixed by #11486.

**Comment by k-takata** [comment]
Fixed by https://github.com/vim/vim/commit/3af982196b1b973e953c35351961f2a96fe34172.
